subprojects/parseq-tracevis-server/src/main/java/com/linkedin/parseq/TracevisServerJarMain.java

```diff
@@ -43,8 +50,34 @@ public static void main(String[] args) throws Exception {
         }
       }
 
-      new TracevisServer(dotLocation, port, base, base, Constants.DEFAULT_CACHE_SIZE, Constants.DEFAULT_TIMEOUT_MS)
-        .start();
+      Pattern pattern = Pattern.compile("[0-9]{4}");
+      if (args.length == 1 || pattern.matcher(args[1]).matches()) { // support http only
+        int httpPort = args.length == 2 ? Integer.parseInt(args[1]) : Constants.DEFAULT_PORT;
+        new TracevisServer(dotLocation, httpPort, base, base, Constants.DEFAULT_CACHE_SIZE, Constants.DEFAULT_TIMEOUT_MS)
+            .start();
+      } else { // support both http and https
+
+        String fileName = args[1];
+        File configFile = new File(fileName);
```

Comment:
We can simply do a `try-with-resource` style to auto-close the file, also avoid creating a File and then a FileInputStream, see example at https://www.mkyong.com/java/java-properties-file-examples/